### ISSUE TITLE
bug(select, autocomplete): strong focus indicator not present for selected option

### ISSUE THREAD

**Issue by zarend** [angular/components#26801]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

For options in a select or autocomplete, strong focus only appears on items that are *not* selected. Indicator does not appear on the selected item.

### Reproduction

Steps to reproduce:
1. https://material2-dev.web.app/select
2. Click "Enable strong focus"
3. Under "ngModel", click on "drink"
4. (Opens a list of drink options "None", "Coke", etc.)
5. Click "Coke"
6. (closes)
7. Click "Coke"

### Expected Behavior

Coke has check icon to indicate selection, gray background and blue focus ring

### Actual Behavior

"Coke" has check icon gray background, but no focus ring.

Pressing up/down to focus other options shows a blue focus ring, but not on "Coke"

<img width="536" alt="image" src="https://user-images.githubusercontent.com/7720245/225767310-8b943c8b-6a63-46ac-bc8d-98733a90f829.png">
<img width="545" alt="image" src="https://user-images.githubusercontent.com/7720245/225767330-2f6531b5-5046-4f16-9b81-c57d08fa79ad.png">

### Environment

- Angular:
- CDK/Material:
- Browser(s):
- Operating System (e.g. Windows, macOS, Ubuntu):
